New docs to teach how to connect and call functions from an Elixir/Phoenix app (#3070)
livebook-dev/livebook: docs/runtime.md

Since your notebook has its own node, it can declare its own Hex package dependencies via `Mix.install/2`.

## Attached runtime

The attached runtime connects to an existing Elixir node managed outside of Livebook, such as a node running a Phoenix application.

```mermaid
graph LR
    subgraph livebook_node["Erlang VM node"]
        A[Livebook]
    end

    subgraph standalone_node["Already running Erlang VM Node"]
        B[attached runtime code evaluator]
        C[Phoenix app]
    end

    A -.-|clusters with| standalone_node
```

When using the attached runtime, your notebook's code cells execute within the same node as the external application. This is similar to attaching an IEx session to a running node.

Since your notebook code runs within the external node's context, your code cells can directly call any function defined in the remote node:

```mermaid
graph LR
    subgraph livebook_node["Erlang VM node"]
        A[Livebook]
    end

    subgraph standalone_node["Already running Erlang VM Node"]
        direction LR

        B[code inside the notebook]
        C[Phoenix app]
    end

>     A -.-|clusters with| standalone_node

Reviewer: ```suggestion
    livebook_node -.-|clusters with| standalone_node
```

    B -.->|direct function call|C
```

> However, your notebook cannot have its own Hex package dependencies, it only has access to what's already loaded in the external node.

Reviewer: ```suggestion
However, your notebook cannot invoke `Mix.install`, it only has access to what's already loaded in the external node.
```
